sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -202,4 +202,25 @@ public static IOChannelFactory getFactory(String spec) throws IOException {
   public static String resolve(String path, String other) throws IOException {
```

Comment:
You should be able to get rid of this method since all previous calls should change over automatically to the other resolve